### ISSUE TITLE
Immersive Block Placing

### ISSUE THREAD

**Issue by hammy275** [hammy3502/immersive-mc#266]
If holding a block and up against the edge of another, place it!

Seems like there would be some false positives though, so I'm thinking on this one more

**Comment by Adidiii** [comment]
I think the current system works fine enough... It would really feel wierd any wouldn't be used all that much, unless you wanna do reach-around placement. If it were implemented, it would just feel like a feature that exists for the sake of existing.

**Comment by hammy275** [comment]
A lot of ImmersiveMC's features are built around increasing immersion at the cost of more confusing controls/time taken. Pulling a lever by moving your hand is a lot more than pressing the use button, for example. This would fall under that, imo

**Comment by Adidiii** [comment]
Yes, but they're still practical, the block placement would need to be perfect if it were to be practical, otherwise, it'd be just annoying. A button is as good as it gets, and is as immersive as it gets without being annoying.

**Comment by hammy275** [comment]
Agreed. It's either going to be really good or not be implemented at all. I don't know if I'm going to be able to get something really good, but this issue will remind me to at least try :smile:

**Comment by floral-qua-floral** [comment]
Perhaps one approach could be a binding which, when held, makes it so that physically hitting a block performs a right-click action, instead of causing mining progress? So for instance, the player holds this button and thwacks the top face of a block in front of them, while they have a block in hand, and their block is placed directly above the one they thwacked. Having this simulate right-clicking rather than making it specific to the action of placing blocks could be nice because it would allow the player to use the same method to do things like stripping logs, waxing copper blocks, or even using some modded tools like wrenches. Although plugging new functionality like this into Vivecraft's existing block-thwacking behavior sounds like it could be tricky or possibly infeasible.

**Comment by Adidiii** [comment]
Isn't that just pressing the A button, only this time you're whacking your hand into the block? It is a viable solution, but I don't think it's immersive since the end result would be your hand inside the block you just placed. I do like the idea for waxing blocks and stripping logs though, if anything comes out of this issue, it should be that interaction.

**Comment by floral-qua-floral** [comment]
The use case I was envisioning with this was to configure a controller that can detect different kinds of grips, such as the Index Knuckles, such that holding the controller in a different grip would be assigned to the keybind. For instance, setting it so that with a closed fist, the player mines blocks, but with an open hand they perform right-click interactions such as placing blocks. It seemed like this could be pretty natural-feeling, since a person naturally opens their hand when putting something down, and would conversely keep their hand closed around the handle of a pickaxe when using it to break something. With controllers that can't detect different kinds of grips, I still figure that thwacking blocks down into the world, albeit with a keybind, could still be more immersive than the point-and-click approach that's currently used.

The player's hand being inside the new block didn't seem particularly concerning to me since that happens when breaking blocks manually, too - it necessarily involves the player's hand or tool clipping into the block. I can see how it'd be a little awkward for block placement, though.
